bug: /recent endpoint crashes gunicorn worker with Permission denied on NFS mount

## Description

The `/recent` endpoint crashes the gunicorn worker when it tries to scan the NFS-mounted data directory, causing the worker to die with `sys.exit(1)`. Users experience the page never loading (spinning indefinitely or 502 errors).

## Root Cause

The `recent_view` function at `app.py:1369` uses `DATA_FOLDER.rglob("*")` to recursively scan all files in the NFS-mounted `/data` directory (5413 items across 5494 items in ~3.5s locally).

The crash occurs inside the gunicorn sync worker when processing the glob. A `Permission denied` error on `/.gunicorn` (gunicorn's pidfile path) propagates up through the worker's request handler, triggering `sys.exit(1)` which kills the worker.

Relevant stack trace:
```
File "/app/app.py", line 1369, in recent_view
    for item in DATA_FOLDER.rglob("*"):
                ~~~~~~~~~~~~~~~~~^^^^^
File "/app/app.py", line 1370, in recent_view
    if not item.is_file():
File "/usr/local/lib/python3.14/site-packages/gunicorn/workers/base.py", line 198, in handle_abort
    sys.exit(1)
SystemExit: 1
```

Simultaneous error in gunicorn logs:
```
[ERROR] Control server error: [Errno 13] Permission denied: '/.gunicorn'
```

## Environment

- **Image**: `ghcr.io/joryirving/miso-gallery:0.1.13@sha256:3704c1aa807e68c0f41e9053abe5eb08e5151b4ec93606cfa17bbfe867724c66`
- **Data mount**: NFS `voyager.internal:/mnt/daena/comfyui/output` mounted at `/data`
- **Runtime user**: uid 1000, gid 100 (non-root)
- **Workers**: 2 sync workers (default in Dockerfile `CMD`)
- **Gunicorn**: 25.3.0

## Symptoms

- Browse to any folder: works fine
- View individual images: works fine  
- Visit `/recent`: page never loads (worker dies before/during rendering)
- Gunicorn continuously respawns dead workers
- Other pages unaffected (they don't use recursive glob)

## Data Directory Permissions

```
/data:  drwxrwxrwt root:root /
/data/.trash:  drwxrwx--- 2 root root   <- only root can access
/data/.thumb_cache:  drwxrwxr-x 1000:100  <- should be accessible
```

## Investigation Notes

- `rglob` completes in ~3.5s in a shell exec (5494 items) — NFS scan itself is not the bottleneck
- The issue is not the `.trash` exclusion — `.trash` is correctly excluded by name
- The error `Permission denied: '/.gunicorn'` is gunicorn trying to write its pidfile at the filesystem root
- The NFS mount shows uid/gid mismatch for some entries (`.trash` owned by `root:root` vs `1000:100` for others)
- This looks like an NFS attribute cache inconsistency — the glob iterates over directory entries that, when later `stat()`d, return stale or inaccessible metadata

## Proposed Fixes (choose one)

### Option A: Defensive exception handling in recent_view (minimal fix)

Wrap the `rglob` loop body in a try/except that skips inaccessible items gracefully instead of crashing the worker:

```python
for item in DATA_FOLDER.rglob("*"):
    try:
        if not item.is_file():
            continue
        # ... rest of loop
    except (OSError, PermissionError) as e:
        continue  # skip inaccessible items
```

### Option B: Shallow glob with os.scandir fallback

Use `os.scandir` for single-level directory traversal with explicit error handling, avoiding the recursive glob on a large NFS directory entirely.

### Option C: Timeout on glob with graceful degradation

Add a timeout/deadline to the glob operation. If it doesn't complete in X seconds, return what was found so far rather than hanging.

## Reproducing

1. Authenticate with miso-gallery
2. Navigate to `/recent`
3. Observe: spinner indefinitely, or 502/500 error
4. Check pod logs: worker crashes with `Permission denied: '/.gunicorn'` followed by worker restart

## Workaround

Rolling restart of the pod temporarily fixes it, but the issue recurs.

--- a/app.py
+++ b/app.py
@@ -1365,42 +1365,46 @@ def recent_view():
         rel_parts = path.relative_to(DATA_FOLDER).parts
         return any(part in excluded_dirs for part in rel_parts)
 
-    try:
-        for item in DATA_FOLDER.rglob("*"):
+    for item in DATA_FOLDER.rglob("*"):
+        try:
+            if not item.is_file() or not is_image(item):
+                continue
+            if is_excluded_from_recent(item):
+                continue
+            if item.name.startswith("."):
+                continue
+
             try:
-                if not item.is_file() or not is_image(item):
-                    continue
-                if is_excluded_from_recent(item):
-                    continue
-                if item.name.startswith("."):
-                    continue
-                try:
-                    mtime = item.stat().st_mtime
-                except OSError:
-                    continue
-
-                rel_path = item.relative_to(DATA_FOLDER).as_posix()
-                date_str = time.strftime("%Y-%m-%d %H:%M", time.localtime(mtime))
-                
-                # Get folder path for navigation
-                folder_path = os.path.dirname(rel_path)
-                folder_url = url_for("index", subpath=folder_path) if folder_path else url_for("index")
-
-                images.append({
-                    "name": item.name,
-                    "rel_path": rel_path,
-                    "url": url_for("view", filename=rel_path),
-                    "thumb": url_for("thumb", filename=rel_path),
-                    "added": date_str,
-                    "size": format_size(item.stat().st_size),
-                    "mtime": mtime,
-                    "folder_url": folder_url,
-                })
-            except (OSError, PermissionError) as e:
+                mtime = item.stat().st_mtime
+                size = item.stat().st_size
+            except (OSError, PermissionError):
                 # Skip items that become inaccessible during NFS attribute cache inconsistency
                 continue
-    except Exception:
-        pass
+
+            try:
+                rel_path = item.relative_to(DATA_FOLDER).as_posix()
+            except (OSError, ValueError):
+                continue
+
+            date_str = time.strftime("%Y-%m-%d %H:%M", time.localtime(mtime))
+
+            # Get folder path for navigation
+            folder_path = os.path.dirname(rel_path)
+            folder_url = url_for("index", subpath=folder_path) if folder_path else url_for("index")
+
+            images.append({
+                "name": item.name,
+                "rel_path": rel_path,
+                "url": url_for("view", filename=rel_path),
+                "thumb": url_for("thumb", filename=rel_path),
+                "added": date_str,
+                "size": format_size(size),
+                "mtime": mtime,
+                "folder_url": folder_url,
+            })
+        except (OSError, PermissionError):
+            # Skip items that raise during iteration
+            continue
 
     images.sort(key=lambda x: x["mtime"], reverse=True)
     images = images[:max_items]

